code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -636,3 +636,21 @@
 	..()
 
 	make_babies()
+
+/mob/living/simple_animal/pet/dog/attack_hand(mob/living/carbon/human/M)
+	. = ..()
+	switch(M.a_intent)
+		if("help")
+			wuv(1,M)
+		if("harm")
+			wuv(-1,M)
+
+/mob/living/simple_animal/pet/dog/proc/wuv(change, mob/M)
+	if(!M || stat == DEAD || !change)
+		return
+	if(change > 0)
+		new /obj/effect/temp_visual/heart(loc)
+		manual_emote("yaps happily!")
```

Comment:
doesn't this need a hologram flag check?

Reply:
this was removed in another PR